@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -434,11 +434,18 @@ export class TextElementsRenderer {
 
         this.updateGlyphDebugMesh();
 
-        this.m_poiRenderer.render(camera, (layerBefore, layerAfter) => {
+        let previousLayer: PoiLayer | undefined;
+        this.m_poiRenderer.update();
+        for (const poiLayer of this.m_poiRenderer.layers) {
             for (const textCanvas of this.m_textCanvases) {
-                textCanvas.render(camera, layerBefore, layerAfter, undefined, false);
+                textCanvas.render(camera, previousLayer?.id, poiLayer.id, undefined, false);
             }
```

Comment:
avoid making this loop if there's still no previousLayer

Reply:
it needs to be called also without a previous layer, in case TextCanvas has layers with lower renderOrder then the lowest in PoiRenderer